File: internship_cs_mar23/jury_solutions/cpp/d.cpp
#include <iostream>
#include <vector>
#include <unordered_set>

using namespace std;

int ans = 0, n;
vector<vector<int>> f;
vector<unordered_set<int>> h;

void rec(int last = 0, int cur = 0) {
	if (h[last].count(cur)) return;
	h[last].insert(cur);
	if (last == n)
		return void(++ans);
	for (int i = 0; i < 6; ++i)
		rec(last + 1, cur + f[last][i]);
}

signed main() {
	cin >> n;
	f.resize(n, vector<int>(6));
	h.resize(n + 1);
	for (int i = 0; i < n; ++i)
		for (int j = 0; j < 6; ++j)
			cin >> f[i][j];
	rec();
	cout << ans;
}

File: internship_cs_mar23/jury_solutions/python/d.py
ans = 0

def rec(last = 0, cur = 0):
	if cur in h[last]: return
	h[last].add(cur)
	if last == n:
	    global ans
	    ans += 1
	    return
	for val in f[last]:
		rec(last + 1, cur + val)

n = int(input())
h = [set() for _ in range(n + 1)]
f = [list(map(int, input().split())) for _ in range(n)]
rec()
print(ans)
